tensorflow/python/eager/benchmarks_test.py

```diff
@@ -1346,6 +1346,12 @@ def benchmark_list_of_zeros_to_np_array(self):
       values.append(array_ops.zeros(shape=(1000,)))
     self._run(lambda: np.array([x.numpy() for x in values]), 1000)
 
+  def benchmark_function_trace(self):
+    @def_function.function
+    def func(x):
+      return x
+    self._run(lambda: (func(x) for x in range(1000)), 30000)
```

Comment:
This repeats `(func(x) for x in range(1000))` 30000 times right? For the latter 29999 iterations, wouldn't it just call a cached function instance because 0-1000 arguments are already called?

Reply:
Yes that’s right, it would cache the values for the function. Would it be better to run a single iteration and increase the range?

Reply:
I'd write the tracing benchmark body as 

```
for _ in range(...):
  tf.function(f)(value)
```

or something like it, as creating a new tf.function object is currently guaranteed to trigger retracing

Reply:
Sounds good to me, we can create a new function on each iteration.